### ISSUE TITLE
Boolean Argument `--trajcontrol` Misinterpreted as `True` When Set to `False`

### ISSUE THREAD

**Issue by MartaYang** [sanweiliti/RoHM#26]
## Description

At [train_trajnet.py#L41](https://github.com/sanweiliti/RoHM/blob/a8c7a3610d4abfd0130b1a4dcd2337ac54298c96/train_trajnet.py#L41C1-L41C113), the `--trajcontrol` argument is defined with `default=False` and `type=bool`, but it always evaluates to `True` when passed via the CLI, even if explicitly set to `False`. This is due to the misuse of `type=bool`, which interprets any non-empty string (e.g., `"False"`) as `True`.

## Steps to Reproduce

1. Define `--trajcontrol` as:
   ```python
   group.add_argument("--trajcontrol", default=False, type=bool, help="If True, finetune trajnet with TrajControl")
   ```

2. Run:
   ```bash
   python train_trajnet.py --trajcontrol False
   ```

3. Observe:
   ```python
   ipdb> print(args.trajcontrol)
   True
   ```

## Suggested Fix

1. Replace `type=bool` with:
   ```python
   type=lambda x: x.lower() in ['true', '1']
   ```
2. Or use `action='store_true'`:
   ```python
   group.add_argument("--trajcontrol", action='store_true', help="If set, finetune trajnet with TrajControl")
   ```

**Comment by sanweiliti** [comment]
You are right, this problem was caused during the code cleaning and reorganizing process, thanks for pointing out the mistake!
